core/domain/classifier_services.py

```diff
@@ -94,10 +97,15 @@ def verify_signature(
         bool. Whether the incoming request is valid.
     """
     secret = None
-    for val in config_domain.VMID_SHARED_SECRET_KEY_MAPPING.value:
-        if val['vm_id'] == oppia_ml_auth_info.vm_id:
-            secret = val['shared_secret_key'].encode('utf-8')
-            break
+    vm_id = secrets_services.get_secret('VM_ID')
+    shared_secret_key = secrets_services.get_secret('SHARED_SECRET_KEY')
+    if (vm_id is None or shared_secret_key is None) and constants.DEV_MODE:
+        vm_id = 'vm_default'
+        shared_secret_key = '1a2b3c4e'
+    assert shared_secret_key is not None
+    assert vm_id is not None
+    if vm_id == oppia_ml_auth_info.vm_id:
+        secret = shared_secret_key.encode('utf-8')
```

Comment:
Hey @seanlip,
So as per our previous discussion, we were planning to add the secret to the `app_dev.yaml` file so that we can get them via `secrets_services.get_secret()` function in the dev mode but after looking at other tests I found out that we are mocking the `get_secret` function in order to test out the things. Please note that when I tried to get the vmid secrets via `secrets_services.get_secret` I was getting None, for this I edited the `app_dev.yaml` file(you can find the changes at the top of the file structure). When I tried to print the value of `os.environ` I was not able to locate the `SECRETS` environment variable.

I have added a check for the dev mode in the changes above in order to pass the default value.

Reply:
Sorry, I'm confused: this is not a test file but you're talking about tests here.

Also, what is your question exactly?

Reply:
Oh, I just wanted to ask if you are fine with the changes above.
Yes, this is not the test file, just wanted to bring your attention that the `get_secret` function is currently not working as expected for the dev environment as instead of getting the secrets from the environment variable we have to mock the function in order to pass the tests. Here in the file, I'm adding a specific check for the dev environment in order to pass the default value but instead, we would be getting the value from the environment variable.
Thanks

Reply:
Ah I see. No, don't do this -- let's not introduce dev-mode conditionals into the main code just to help the tests. Do explicit mocking for the tests if needed.

Reply:
But this condition is not only for the tests, if I remove this then we will get an error in the local development server when we try to get the vmid secrets. What should we do in that case?

Reply:
What do we currently do for other secrets?

Generally we should have a process for handling secrets in dev mode that actually works. Is this code actually exercisable in dev mode when using the server normally? If so, I agree with you that we need to find a way, but if not, maybe we can file an issue for that and leave it to later.

If we do need to fix it, perhaps search and read up about how cloud secrets are supposed to be handled in dev mode -- we can't be the first folks to run into this issue.

Reply:
I took a look at other secrets, we are simply using them via `secrets_services.get_secret()` without any conditions.
We can close this discussion as I was able to access the secrets when the dev server was on. It now makes sense why we use the swapping for the `get_secret` function in the tests as we cannot really access the environment variables when running the tests.

I have updated the tests and the functionality works fine.